core/trino-main/src/main/java/io/trino/server/CoordinatorDiscoveryModule.java

```diff
@@ -23,7 +23,7 @@
     @Override
     protected void setup(Binder binder)
     {
-        if (buildConfigObject(ServerConfig.class).isCoordinator() &&
+        if (buildConfigObject(ServerConfig.class).isCoordinator() ||
```

Comment:
is `discovery-server.enabled=false` now ignored?

Reply:
on coordinator, yes

Reply:
I think it would be nice to honor `discovery-server.enabled=false`, or to throw configuration validation  if it is supposed to be ignored